### PR TITLE
SSMetrics

### PR DESCRIPTION
**Pull request by alexkar598**
# Document the changes in your pull request

Records performance data on the game
Heavily inspired by @AffectedArc07's code in ParadiseSS13/Paradise#16549

### CODE REVIEW

**Comment by alexkar598** [comment]
image
image

File: code/controllers/subsystem/metrics.dm [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	if (length(SSmetrics.queue) > threshold)
+		RUSTG_CALL(RUST_G, "influxdb2_publish")(json_encode(SSmetrics.queue), CONFIG_GET(string/metrics_api), CONFIG_GET(string/metrics_token))
+		SSmetrics.queue = list()
+
+/datum/controller/subsystem/metrics_publish/proc/profile_publish()
+	if ((last_profile_publish + profile_publish_interval) < REALTIMEOFDAY)
+		last_profile_publish = REALTIMEOFDAY
+		var/data = world.Profile(PROFILE_REFRESH, "json")
+		world.Profile(PROFILE_CLEAR)
+		message_admins(data)
```
Comment (adamsong) [inline code/controllers/subsystem/metrics.dm]:
```suggestion
```

**Review by adamsong — CHANGES_REQUESTED** [review]
Pls no spam message_admins

**Review by adamsong — DISMISSED** [review]
Update rust_g.dll

**Review by adamsong — APPROVED** [review]

**Comment by adamsong** [comment]
"all should be good now" - alexkar

**Review by ToasterBiome — APPROVED** [review]

**Pull request merged**